src/MSBuildLocator/DotNetSdkLocationHelper.cs

```diff
@@ -102,14 +102,17 @@ private static IEnumerable<string> GetDotNetBasePaths(string workingDirectory)
                 string filePath = Path.Combine(dir, exeName);
                 if (File.Exists(Path.Combine(dir, exeName)))
                 {
-                    dotnetPath = filePath;
-                    break;
+                    dotnetPath = Path.GetDirectoryName(isWindows ? filePath : realpath(filePath) ?? filePath);
+                    if (File.Exists(dotnetPath))
```

Comment:
Would you want to validate the file path `realpath(filePath)` exists instead of just the directory?